src/mango/test/25-beginswith-test.py

```diff
@@ -0,0 +1,128 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import copy
+import mango
+
+DOCS = [
+    {"_id": "aaa", "name": "Jimi", "location": "AUS", "age": 27},
+    {"_id": "abc", "name": "Eddie", "location": "AND", "age": 65},
+    {"_id": "bbb", "name": "Harry", "location": "CAN", "age": 21},
+    {"_id": "ccc", "name": "Eddie", "location": "DEN", "age": 37},
+    {"_id": "ddd", "name": "Jones", "location": "ETH", "age": 49},
+]
+
+
+def to_utf8_bytes(list):
+    return [x.encode() for x in list]
+
+
+class BeginsWithOperator(mango.DbPerClass):
+    def setUp(self):
+        self.db.recreate()
+        self.db.save_docs(copy.deepcopy(DOCS))
+        self.db.create_index(["location"])
+        self.db.create_index(["name", "location"])
+
+    def get_mrargs(self, selector, sort=None):
+        explain = self.db.find(selector, sort=sort, explain=True)
+        return explain["mrargs"]
+
+    def assertDocIds(self, user_ids, docs):
+        user_ids_returned = list(d["_id"] for d in docs)
+        user_ids.sort()
+        user_ids_returned.sort()
+        self.assertEqual(user_ids, user_ids_returned)
+
+    def test_basic(self):
+        docs = self.db.find({"location": {"$beginsWith": "A"}})
+
+        self.assertEqual(len(docs), 2)
+        self.assertDocIds(["aaa", "abc"], docs)
+
+    def test_json_range(self):
+        mrargs = self.get_mrargs({"location": {"$beginsWith": "A"}})
+
+        self.assertEqual(mrargs["start_key"], ["A"])
+        end_key_bytes = to_utf8_bytes(mrargs["end_key"])
+        self.assertEqual(end_key_bytes, [b"A\xef\xbf\xbd", b"<MAX>"])
+
+    def test_compound_key(self):
+        selector = {"name": "Eddie", "location": {"$beginsWith": "A"}}
+        mrargs = self.get_mrargs(selector)
+
+        self.assertEqual(mrargs["start_key"], ["Eddie", "A"])
+        end_key_bytes = to_utf8_bytes(mrargs["end_key"])
+        self.assertEqual(end_key_bytes, [b"Eddie", b"A\xef\xbf\xbd", b"<MAX>"])
+
+        docs = self.db.find(selector)
+        self.assertEqual(len(docs), 1)
+        self.assertDocIds(["abc"], docs)
+
+    def test_sort(self):
+        selector = {"location": {"$beginsWith": "A"}}
+        mrargs = self.get_mrargs(selector, sort=["location"])
+
+        self.assertEqual(mrargs["start_key"], ["A"])
+        end_key_bytes = to_utf8_bytes(mrargs["end_key"])
+        self.assertEqual(end_key_bytes, [b"A\xef\xbf\xbd", b"<MAX>"])
+        self.assertEqual(mrargs["direction"], "fwd")
+
+    def test_sort_desc(self):
+        selector = {"location": {"$beginsWith": "A"}}
+        mrargs = self.get_mrargs(selector, sort=[{"location": "desc"}])
+
+        start_key_bytes = to_utf8_bytes(mrargs["end_key"])
```

Comment:
I think this is not right: either the variable name (`start_key_bytes`) or the corresponding field name (`"end_key"`) is wrong here.